salt/modules/nilrt_ip.py

```diff
@@ -258,45 +259,102 @@ def _remove_quotes(value):
     return value
 
 
-def _get_requestmode_info(interface):
+def _load_config(section, options, default_value='', file=INI_FILE):
+    """
+    Get values for some options and a given section from a config file.
+
+    :param section: Section Name
+    :param options: List of options
+    :param default_value: Default value if an option doesn't have a value. Default is empty string.
+    :param file: config file. Default is INI_FILE.
+    :return:
+    """
+    results = {}
+    if not options:
+        return results
+    with salt.utils.files.fopen(file, 'r') as config_file:
+        config_parser = configparser.RawConfigParser(dict_type=CaseInsensitiveDict)
+        config_parser.readfp(config_file)
+        for option in options:
+            if six.PY2:
+                results[option] = _remove_quotes(config_parser.get(section, option)) \
+                    if config_parser.has_option(section, option) else default_value
+            else:
+                results[option] = _remove_quotes(config_parser.get(section, option, fallback=default_value))
+
+    return results
+
+
+def _get_request_mode_info(interface):
     '''
     return requestmode for given interface
     '''
-    with salt.utils.files.fopen(INI_FILE, 'r') as config_file:
-        config_parser = configparser.RawConfigParser(dict_type=CaseInsensitiveDict)
-        config_parser.read_file(config_file)
-        link_local_enabled = '' if not config_parser.has_option(interface, 'linklocalenabled') else \
-            int(_remove_quotes(config_parser.get(interface, 'linklocalenabled')))
-        dhcp_enabled = '' if not config_parser.has_option(interface, 'dhcpenabled') else \
-            int(_remove_quotes(config_parser.get(interface, 'dhcpenabled')))
-
-        if dhcp_enabled == 1:
-            return 'dhcp_linklocal' if link_local_enabled == 1 else 'dhcp_only'
-        else:
-            if link_local_enabled == 1:
-                return 'linklocal_only'
-            if link_local_enabled == 0:
-                return 'static'
+    settings = _load_config(interface, ['linklocalenabled', 'dhcpenabled'])
+    link_local_enabled = int(settings['linklocalenabled'])
+    dhcp_enabled = int(settings['dhcpenabled'])
+
+    if dhcp_enabled == 1:
+        return 'dhcp_linklocal' if link_local_enabled == 1 else 'dhcp_only'
+    else:
+        if link_local_enabled == 1:
+            return 'linklocal_only'
+        if link_local_enabled == 0:
+            return 'static'
 
     # some versions of nirtcfg don't set the dhcpenabled/linklocalenabled variables
     # when selecting "DHCP or Link Local" from MAX, so return it by default to avoid
     # having the requestmode "None" because none of the conditions above matched.
     return 'dhcp_linklocal'
 
 
+def _get_adapter_mode_info(interface):
+    '''
+    return adaptermode for given interface
+    '''
+    mode = _load_config(interface, ['mode'])['mode']
```

Comment:
I didn't put `default_value = 'tcpip'` because the file may be corrupted and mode should be one of these values:
* **disabled**
* **ethercat**
* **tcpip** (default value)